agent/engine/task_manager.go

```diff
@@ -439,6 +448,39 @@ func (mtask *managedTask) handleContainerChange(containerChange dockerContainerC
 		mtask.Arn, container.Name, mtask.GetDesiredStatus().String())
 }
 
+// handleResourceChange updates a resource's known status
+func (mtask *managedTask) handleResourceChange(res taskresource.TaskResource,
+	status taskresource.ResourceStatus,
+	err error) {
+	// locate the resource
+	found := mtask.isResourceFound(res)
+	if !found {
+		seelog.Criticalf("Managed task [%s]: state error; invoked with another task's resource [%s]!",
+			mtask.Arn, res.GetName())
+		return
+	}
+
+	currentKnownStatus := res.GetKnownStatus()
+
+	if status <= currentKnownStatus {
+		seelog.Infof("Managed task [%s]: redundant resource state change. %s to %s, but already %s",
+			mtask.Arn, res.GetName(), res.StatusString(status), res.StatusString(currentKnownStatus))
+		return
+	}
+
+	if err == nil {
+		res.SetKnownStatus(status)
+		return
+	}
+	seelog.Debugf("Managed task [%s]: error while transitioning resource %s to %s: %v",
```

Comment:
Just for my own understanding, what's the logic here? There's an error while transitioning the resource. But, you only set task's desired status to STOPPED is resource's status is equal to it's steady state? Why is that?

Reply:
once we move the task's desired status to STOPPED, we eventually move every container and resources' desired status to STOPPED [here](https://github.com/aws/amazon-ecs-agent/pull/1332/files#diff-e008975819a900a8494cf91881a8304aR872). I can add a comment.

Reply:
Hmm, I still didn't understand why task's desired status should be conditionally set to STOPPED. Seems like irrespective of resource's status, task's desired status should be set to STOPPED.

Reply:
right now, errors during steady state is the only place, where we want to move the task to STOPPED. there may be other states in the future where we might want to handle things differently.  hence the condition check.